src/Object/Datasource/DatasourceProperties.php

```diff
@@ -15,22 +15,22 @@ class DatasourceProperties extends BaseApiResource
      * @return mixed
      */
     public function getProperties() {
-        return $this->data[self::FIELD_PROPERTIES];
+        return $this->data[static::FIELD_PROPERTIES];
     }
 
     /**
      * @param $properties
      * @return $this
      */
     public function setProperties($properties) {
-        $this->{self::FIELD_PROPERTIES} = $properties;
+        $this->{static::FIELD_PROPERTIES} = $properties;
 
         return $this;
     }
     
     /**
-     * A DatasourceProprerty always exists because the PUT creates it if it does not exist or
-     * updates it if it existss.
+     * A DatasourceProperty always exists because the PUT creates it if it does not exist or
+     * updates it if it exists.
      * @return bool
      */
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/sonarqube/coding_rules#rule_key=php%3AS1135)